File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // type your code here
  let newString = '' 
  for (i = str.length -1; i >=0; i--){
    newString = newString + str[i]
  }
  return newString
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

// create an empty array (empty string) to store the reversed letters
// iterate over the original string, beginning at the end, and work backwards
// add each element of the original string in reverse order to the new string
// return the empty string when finished iterating 

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  # type your code in here
  new_string = ''
  str.chars.each do |letter|
    new_string = letter + new_string
  end
  new_string
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution

# create an empty string
# for each character in the original string, prepend it to the new string
# return the new string